### DISCUSSION TITLE
scheduleJobs atomicity

### DISCUSSION THREAD

**Discussion by gtaylor44** [quartznet/quartznet#675] [General] [answered]
I can't see anywhere in the documentation specifying if scheduleJobs is wrapped in a transaction or not.

scheduleJobs(IReadOnlyDictionary<IJobDetail, IReadOnlyCollection<ITrigger>> triggersAndJobs, boolean replace)

I'm faced with a potential problem where a user updates their timezone. This invalidates all their cron expressions which need be rescheduled. Either everything updates or nothing updates is expected behaviour.

Will scheduleJobs rollback if i'm using SQL Server as data store and one of the triggers or jobs fails to be deleted or inserted?

**Comment by lahma** [comment, marked as answer]
Here's the logic you are referring, line 801:

https://github.com/quartznet/quartznet/blob/5d14cbd4af7d7c82b274282b7d01b746560a7ad9/src/Quartz/Core/QuartzScheduler.cs#L753-L804

Which in turn calls JobStoreSupport:

https://github.com/quartznet/quartznet/blob/5d14cbd4af7d7c82b274282b7d01b746560a7ad9/src/Quartz/Impl/AdoJobStore/JobStoreSupport.cs#L1134-L1154

Which in turn flows all the way to running code under lock:

https://github.com/quartznet/quartznet/blob/5d14cbd4af7d7c82b274282b7d01b746560a7ad9/src/Quartz/Impl/AdoJobStore/JobStoreSupport.cs#L3639-L3717

So the code that handles insert for the batch is running under ADO.NET transaction.

**Comment by gtaylor44** [comment]
That's awesome. Thank you very much.
